### ISSUE TITLE
When will this plugin be available?

### ISSUE THREAD

**Issue by adelinn** [bcc-code/btvlive-kodi#1]

**Comment by Ghostbird** [comment]
Currently it's waiting for response from the BrunstadTV team regarding the clientId I need to connect to BCC SSO.

**Comment by Ghostbird** [comment]
I had received a reply last week that I should contact a different, non-technical, department. I have e-mailed them today.

**Comment by Ghostbird** [comment]
I have sent a reminder e-mail today, as I have had no response yet.

**Comment by KillerX** [comment]
I'm poking from my side as well...

**Comment by Ghostbird** [comment]
@KillerX sent me a request for the source as it is, for an internal code audit. I've submitted it today.

**Comment by Ghostbird** [comment]
It seems we're going to publish this soon. Questions related to the code audit revealed a substantial shortcoming on the BrunstadTV side that would compromise the security of this add-on if we continue with the current set-up. However I have heard this will likely not be fixed before 2022.

**Comment by KillerX** [comment]
Just to elaborate on what @Ghostbird wrote:
He used a login flow that was initially recommended from my side, but this particular flow seems to enable anyone with access to a "public" clientID to preform a login with no involvement of any secrets, meaning that we would essentially completely open to login system. This is currently not wanted due to some mentioned shortcomings that are being worked on.

In short there are 2 options that we are investigating:

* I don't understand the flow correctly and there is actually something we can do to limit things (probably, since smarter people than me have designed the system). Here we have another person looking into it
* I understood the flow correctly and we will likely need to wait for us to fix the underlying limit that we have currently with the auth provider. This is in progress but it will take a bit to complete.

Either way I am working on getting approval to release this. Thanks for being patient, it's our first rodeo with a "3rd party" app integrating into our system and there are surprisingly many things internally that need to be resolved/adjusted for this to work. Hopefully when this is resolved any new systems will have an easier time!

**Comment by Ghostbird** [comment]
Hey Matjaz,

> meaning that we would essentially completely open to login system

You guys should realise that you *are* such a system already. Any developer
can grab the clientId from the BMM webapp, the BMM or BrunstadTV
android/iOS client, the myshare webapp etc. and build their own app with it.

The OAuth2 standard is very aware of this. For example, when you have an
app that's running client-side (webapp, desktop app, mobile app) you should
not use a client secret, because it is meaningless. It isn't a secret if
you publish it to every customer's device.
The protection against misuse in OAuth2 consists of CORS restrictions for
webapps, and most importantly, a consent screen with client specific
information. Hence it is *essential to not share client IDs.*
The client information upon login is the users' primary protection against
malicious apps.

Sincerely,

Gijsbert ter Horst

**Comment by KillerX** [comment]
> You guys should realise that you *are* such a system already. 

That is correct, but as far as I understand not for the reasons you mention.

> Any developer can grab the clientId from the BMM webapp, the BMM or BrunstadTV android/iOS client, the myshare webapp > etc. and build their own app with it.

This is slightly incorrect. The main thing that prevents just anyone from doing it on a large scale is the callback url component of the normal flows. Since the flow will only redirect to whitelisted callback URLs, the direct hijacking possibilities are somewhat limited. For web applications I would say it is self explanatory, while iOs and Android allow a domain to designate only specific apps to allow opening deep links (again preventing arbitrary app to intercept the data), like here: https://brunstad.tv/.well-known/assetlinks.json, https://brunstad.tv/apple-app-site-association

> The protection against misuse in OAuth2 consists of CORS restrictions for
> webapps, and most importantly, a consent screen with client specific
> information. Hence it is *essential to not share client IDs.*
> The client information upon login is the users' primary protection against
> malicious apps.

Very much agreed, specifically the last part.

**Comment by Ghostbird** [comment]
Ah dang, I completely forgot about the registered callback URIs for webapps, that's a good protection indeed. I didn't actually know about those iOS and Android mechanisms. I looked into the iOS and Android app deep link mechanism, I don't think it's applicable to this Kodi plugin, but it might be useful in other situations. Does the deep link mechanism apply to the login redirect URI though? The apps I've worked with used a custom URI scheme, but AFAIK there's no stopping a malicious app from supporting the same URI scheme. Maybe the custom URI scheme should in such cases be replaced with a deep link instead?

I see now that the current situation is indeed a bit different than any other app you've connected to BCC SSO before. Which makes it much more important here to have clear and distinct client information here, since it seems to be the only line of defence against abuse.

**Comment by Ghostbird** [comment]
@KillerX Any update from your side?

**Comment by KillerX** [comment]
As a matter of fact, yes!
We now have possibility to provide you with a proper ID. There are still some things we need to iron out, but the technical issues should not stand in the way anymore. I'll try to get back to you by Wednesday.

**Comment by Ghostbird** [comment]
@KillerX Coming Wednesday might be a _bit_ short notice. However I can offer you next Wednesday. Is that ok?

**Comment by KillerX** [comment]
I'm waiting internally for a document that needs signature. This now the last (to my knowledge) obstacle.

**Comment by Ghostbird** [comment]
Any update?
